tool/planet/etcd.go

```diff
@@ -412,23 +504,100 @@ func getBaseEtcdDir(version string) string {
 }
 
 func etcdRestore(file string) error {
-	ctx := context.TODO()
-	log.Info("Restoring backup to temporary etcd")
+	log.Info("Initializing new etcd database")
+
+	datadir, err := getEtcdDataDir()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// To be re-entrant, shutdown the etcd server if it's already running
+	err = etcdDisable(true, false)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// For the restore operation to be re-entrant, delete the data directory in-case the step is being re-run
+	err = os.RemoveAll(datadir)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	err = etcdEnable(true, "")
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	etcdConf := etcdconf.Config{
+		Endpoints: []string{DefaultEtcdUpgradeEndpoints},
+		KeyFile:   DefaultEtcdctlKeyFile,
+		CertFile:  DefaultEtcdctlCertFile,
+		CAFile:    DefaultEtcdctlCAFile,
+	}
+	client, err := etcdConf.NewClient()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// wait for the temporary etcd instance to complete startup
+	log.Info("Waiting for etcd initialization to complete")
+	err = waitEtcdHealthyTimeout(context.TODO(), 2*time.Minute, client)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// stop etcd now that it's DB is initialized but empty, to run offline restore to the empty database
+	log.Info("Etcd initialization complete, stopping")
+	err = etcdDisable(true, false)
```

Comment:
Nit: To help the reader figure out what "true" and "false" are I would either create variables and user them here (i.e. `var confirmed = true, etcdDisable(confirmed, ...)` or whatever they are), or convert them into a struct. Same for `etcdEnable`.